include/diy/master.hpp

```diff
@@ -491,14 +513,17 @@ void
 diy::Master::
 unload_incoming(int gid)
 {
-  IncomingQueuesRecords& in_qrs = incoming_[gid];
-  for (InQueueRecords::iterator it = in_qrs.records.begin(); it != in_qrs.records.end(); ++it)
+  for (IncomingRoundMap::iterator round_itr = incoming_.begin(); round_itr != incoming_.end(); ++round_itr)
   {
-    QueueRecord& qr = it->second;
-    if (queue_policy_->unload_incoming(*this, it->first, gid, qr.size))
+    IncomingQueuesRecords& in_qrs = round_itr->second.map[gid];
```

Comment:
This should be `rount_itr->second.map.find(gid)`, and then skip if it doesn't exist. We don't want to be creating new records in here, even if they are empty.